File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcCurve/IfcPolynomialCurve/IfcPolynomialCurve.vb
Public Class IfcPolynomialCurve Inherits IfcCurve
    Public Position As IfcPlacement
    Public CoefficientsX As List(Of IfcReal)
    Public CoefficientsY As List(Of IfcReal)
    Public CoefficientsZ As List(Of IfcReal)

    ' === WHERE CLAUSES ===
    ' CorrectPositionDim : ((Position.Dim=2) AND (NOT EXISTS(CoefficientsZ))) OR (Position.Dim=3)
    ' ValidCoefficients : (EXISTS(CoefficientsX) AND EXISTS(CoefficientsY)) OR (EXISTS(CoefficientsX) AND EXISTS(CoefficientsZ)) OR (EXISTS(CoefficientsY) AND EXISTS(CoefficientsZ)) OR (EXISTS(CoefficientsX) AND EXISTS(CoefficientsY) AND EXISTS(CoefficientsZ))
End Class

File: src/main/resources/output/IFC4X3_java/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcCurve/IfcPolynomialCurve/IfcPolynomialCurve.java
public class IfcPolynomialCurve extends IfcCurve {
    public IfcPlacement Position;
    public List<IfcReal> CoefficientsX;
    public List<IfcReal> CoefficientsY;
    public List<IfcReal> CoefficientsZ;

    // === WHERE CLAUSES ===
    // CorrectPositionDim : ((Position.Dim=2) AND (NOT EXISTS(CoefficientsZ))) OR (Position.Dim=3)
    // ValidCoefficients : (EXISTS(CoefficientsX) AND EXISTS(CoefficientsY)) OR (EXISTS(CoefficientsX) AND EXISTS(CoefficientsZ)) OR (EXISTS(CoefficientsY) AND EXISTS(CoefficientsZ)) OR (EXISTS(CoefficientsX) AND EXISTS(CoefficientsY) AND EXISTS(CoefficientsZ))
}
